test/jackdaw/streams_test.clj

```diff
@@ -198,20 +198,38 @@
           (System/setOut std-out)))))
 
   (testing "through"
-    (let [topic-a (mock/topic "topic-a")
-          topic-b (mock/topic "topic-b")
-          topic-c (mock/topic "topic-c")
-          driver (mock/build-driver (fn [builder]
-                                      (-> builder
-                                          (k/kstream topic-a)
-                                          (k/through topic-b)
-                                          (k/to topic-c))))
-          publish (partial mock/publish driver topic-a)]
-
-      (publish 1 1)
+    (testing "without partitions"
+      (let [topic-a (mock/topic "topic-a")
+            topic-b (mock/topic "topic-b")
+            topic-c (mock/topic "topic-c")
+            driver (mock/build-driver (fn [builder]
+                                        (-> builder
+                                            (k/kstream topic-a)
+                                            (k/through topic-b)
+                                            (k/to topic-c))))
+            publish (partial mock/publish driver topic-a)]
+
+        (publish 1 1)
+
+        (is (= [[1 1]] (mock/get-keyvals driver topic-b)))
+        (is (= [[1 1]] (mock/get-keyvals driver topic-c))))))
+    (testing "with partition"
+      (let [topic-a (mock/topic "topic-a")
+            topic-b (assoc (mock/topic "topic-b") :partition-fn (fn [key value partition-count]
+                                                                  10))
+            topic-c (mock/topic "topic-c")
+            driver (mock/build-driver (fn [builder]
+                                        (-> builder
+                                            (k/kstream topic-a)
+                                            (k/through topic-b)
+                                            (k/to topic-c))))
+            publish (partial mock/publish driver topic-a)]
+
+        (publish 1 1)
+
+        (is (= [[1 1 10]] (mock/get-keyvals driver topic-b)))
+        (is (= [[1 1]] (mock/get-keyvals driver topic-c))))))
```

Comment:
`jackdaw.streams.mock/get-keyvals` might be doing the wrong thing here. When more than a key and value is returned, the result should probably be a map. See `jackdaw.client/poll` for the expected behavior.

Reply:
This was an intermediary push. Can you check its current state